File: soyeon/[3] Programmers/카펫.java
class Solution {
	public int[] solution(int brown, int yellow) {
		int[] answer = new int[2];

		int all = brown + yellow;
		int row = 3;
		while (true) {
			if (all % row == 0) {
				int col = all / row;

				if (yellow == (row - 2) * (col - 2)) {
					answer[0] = Math.max(row, col);
					answer[1] = Math.min(row, col);
					break;
				}
			}
			row++;
		}

		return answer;
	}
}

File: kuyhochung/5. 프로그래머스/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    int total = brown + yellow;
    vector<int> answer;
    for (int i = 3;; i++) {
        int temp = total / i;
        if (total % i == 0 && temp >= i && temp*2 + i*2 == brown + 4) {
            answer.push_back(temp);
            answer.push_back(i);
            break;
        }
    }
    return answer;
}